tools/upload-grid.mjs

```diff
@@ -0,0 +1,83 @@
+import fs from 'fs';
+import { getEnvKey } from './lib/envkeys.mjs';
+import { fetchProject, assignSessionsToSlotAndRoom } from './lib/project.mjs'
+import { validateSession } from './lib/validate.mjs';
+import seedrandom from 'seedrandom';
+
+function readconfig({ filename }) {
+   if (filename) {
+      let content = fs.readFileSync(filename).toString();
+      let regexp = /<pre id="data">[^<]+<\/pre>/s;
+      let data = content.match(regexp)[0].match(/\[\{.*\}\]/)[0];
+      return(Array.from(JSON.parse(data)));
```

Comment:
`Array.from` should not be needed. `JSON.parse` should already give you an array.